File: grind75/Jan15/198.go
func rob(nums []int) int {
    n := len(nums)
    if n == 1 { return nums[0] }
    if n == 2 { return max(nums[0], nums[1]) }
    dp := make([]int, n)
    dp[0], dp[1] = nums[0], max(nums[0], nums[1])
    ret := 0
    for i := 2; i < n; i++ {
        dp[i] = max(nums[i]+dp[i-2], dp[i-1])
        ret = max(ret, dp[i])
    }
    return ret
}   

File: grind75/Jan15/198.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [0] * n
        for i, num in enumerate(nums):
            if i == 0:
                dp[i] = num
            elif i == 1:
                dp[i] = max(dp[i-1], num)
            else:
                dp[i] = max(dp[i-1], dp[i-2] + num)
        return dp[-1]
